coffeescript-loader/test.js

```diff
@@ -1,26 +1,32 @@
 import { ok } from 'assert';
 import { spawn } from 'child_process';
 import { execPath } from 'process';
+import { fileURLToPath, URL } from 'url';
 
 
 // Run this test yourself with debugging mode via:
 // node --inspect-brk --experimental-loader ./loader.js ./fixtures/esm-and-commonjs-imports.coffee
 
 const child = spawn(execPath, [
   '--experimental-loader',
-  './loader.js',
-  './fixtures/esm-and-commonjs-imports.coffee'
+  fileURLToPath(new URL('./loader.js', import.meta.url).href),
+  fileURLToPath(new URL('./fixtures/esm-and-commonjs-imports.coffee', import.meta.url).href),
 ]);
-
 let stdout = '';
 child.stdout.setEncoding('utf8');
 child.stdout.on('data', (data) => {
   stdout += data;
 });
+let stderr = '';
+child.stderr.setEncoding('utf8');
+child.stderr.on('data', (data) => {
+  stderr += data;
+});
 
 child.on('close', (code, signal) => {
-  stdout = stdout.toString();
   ok(stdout.includes('Hello from CoffeeScript', 'Main entry transpiles'));
   ok(stdout.includes('HELLO FROM ESM'), 'ESM import transpiles');
-  ok(stdout.includes('Hello from CommonJS!'), 'CommonJS import transpiles');
+
+  // this relies on a named export from CommonJS, which is not supposed to work
```

Comment:
So is this a TODO? Could we spell out more clearly what is supposed to happen, and mention that this is a bug/unimplemented feature?

Reply:
I'm honestly not sure what is supposed to happen—I expect this to work now and before. But it doesn't work then or now. I'm afraid to open a can of worms investigating this, but if it's supposed to work, I (biting my tongue so hard) think it should maybe block nodejs/node#37468 (depending on how pervasive the usage of this is).

What currently happens is the `require.extension` works, the transformed source disappears into the CJS loader (I confirmed it does get passed it), and then an empty object comes out the other side.

Reply:
There’s no need to block a PR because it doesn’t fix an existing bug, especially not for an experimental feature. I think all we need to do at this point is add a comment here explaining something like “known issue: blah blah” with the clearest explanation you can write, and we’ll come back to it later. I just don’t want to forget about this or have this code in here and someone like you comes along and thinks that it should be working, or that it did work in some older version of Node but got broken somehow more recently.